mantidproject/mantid
# Fix legend order changing after changing curve settings

Report to: James Lord and @smk78

**Description of work.**
This PR makes it so the legend handles are set manually when adding a legend to plots, so that the order of the curves stays the same.

**To test:**
Run
```
from mantid.simpleapi import *
import matplotlib.pyplot as plt

ws = CreateSampleWorkspace()

fig, axes = plt.subplots(subplot_kw={'projection': 'mantid'})

for i in range(1,5):
    axes.plot(ws, specNum=i)

axes.legend().set_draggable(True)

plt.show()
```

Open figure options, go to the Curves tab, change some settings for a curve and check that its position in the legend stays the same.

Fixes #26998 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
